gen/solid.hoon

```diff
@@ -10,59 +10,85 @@
 ::::
   !:
 :-  %say
-|=  $:  {now/@da eny/@uvJ bec/beak}
-        {arg/$@(~ {top/path ~}) dub/_|}
+|=  $:  [now=@da eny=@uvJ bec=beak]
+        arg=$@(~ [top=path ~])
+        dub/_|
     ==
-?~  arg  $(arg ~[top=`path`/(scot %p p.bec)/[q.bec]/(scot %da now)/sys])
-::
 :-  %noun
-=+  pax=`path`(weld top.arg `path`[%hoon ~])
-=+  arp=`path`(weld top.arg `path`[%arvo ~])
+::  sys: root path to boot system, `/~me/[desk]/now/sys`
+::
+=/  sys=path
+  ?^  arg  top.arg
+  /(scot %p p.bec)/[q.bec]/(scot %da now)/sys
+::
+=/  compiler-path  (weld sys /hoon)
+=/  arvo-path      (weld sys /arvo)
 ~&  %solid-start
-=+  txt=.^(@t %cx (weld pax `path`[%hoon ~]))
-=+  rax=.^(@t %cx (weld arp `path`[%hoon ~]))
-=+  ^=  ken
-    =-  ?:(?=(%& -.res) p.res (mean (flop p.res)))
-    ^=  res  %-  mule  |.
-    ~&  %solid-loaded
-    =+  gen=(rain pax txt)
+=/  compiler-src  .^(@t %cx (weld compiler-path /hoon))
+=/  arvo-src      .^(@t %cx (weld arvo-path /hoon))
+=/  arvo-formula
+  =;  res
```

Comment:
I know it was already this way, but why do we bother calling +mule here? All we do with the error is print and die, which I would assume is the same as what would happen if we ran it without an extra virtualization layer.

Reply:
Agreed, there's no reason to mule here.